components/permissions/contexts/brave_ethereum_permission_context.cc

```diff
@@ -47,7 +47,10 @@ BraveEthereumPermissionContext::BraveEthereumPermissionContext(
 BraveEthereumPermissionContext::~BraveEthereumPermissionContext() = default;
 
 bool BraveEthereumPermissionContext::IsRestrictedToSecureOrigins() const {
-  return true;
+  // For parity with Crypto Wallets and MM we should allow a permission prompt
+  // to be shown for HTTP sites. Developers often use localhost for development
+  // for example.
```

Comment:
FYI i think you can allow localhost without allowing all HTTP. according to the secure contexts spec, localhost should be considered secure. https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts

are there other HTTP sites that require this functionality?

Reply:
Our current wallet Crypto Wallets and MM both expose window.ethereum to HTTP sites, not only localhost. 
So this is for webcompat / parity with that.
I don't know of which dapps are popular or not, but it's very feasible that users use ganache or a node on their lan, or use a dapp that is only on HTTP.

Reply:
Just to add a bit more info...
`window.ethereum` will be exposed wether or not a site is HTTP before and after this PR both. Also a site can use `window.ethereum` (including making requests to ethereum over HTTPS even if the site is HTTP) without permissions.

This PR allows an HTTP site to request permissions though.
The permission gives the site access to:
- The selected user allowed address
- Adding and signing transactions for that address (requires user approval still)
- Signing messages for that address (requires user approval still)